src/breakupModel/simulation/Breakup.cpp

```diff
@@ -1,110 +1,101 @@
-        const double mass = area / areaToMassRatio;
-
-        //Finally set every value in the satellite
-        satIt->setAreaToMassRatio(areaToMassRatio);
-        satIt->setArea(area);
-        satIt->setMass(mass);
-
-        //Stop calculating further values if the produced mass already exceeds the input mass
-        //and erases the elements which are no longer needed
-        _outputMass += mass;
-        if (_outputMass > _inputMass) {
-//            spdlog::warn("The simulation reduced the number of fragments because the mass budget was exceeded. "
-//                         "In other words: The random behaviour has produced heavier fragments");
-            _outputMass -= mass;
-            _output.erase(satIt, _output.end());
-            break;
-        }
+        std::get<3>(tuple) = std::get<2>(tuple) / std::get<1>(tuple);
+    });
+    //Enforce Mass Conservation if the output mass is greater than the input mass
+    _outputMass = std::accumulate(_output._mass.begin(), _output._mass.end(), 0.0);
+    size_t oldSize = _output.size();
+    size_t newSize = _output.size();
+    while (_outputMass > _inputMass) {
+        _outputMass -= _output._mass.back();
+        newSize -= 1;
+        _output._mass.pop_back();
+    }
+    if (oldSize != newSize) {
+        spdlog::warn("The simulation reduced the number of fragments because the mass budget was exceeded. "
+                     "In other words: The random behaviour has produced heavier fragments");
+        _output.resize(newSize);
     }
 }
 
 void Breakup::deltaVelocityDistribution(double factor, double offset) {
-    std::for_each(_output.begin(),
-                  _output.end(),
-                  [&](Satellite &sat) {
-                      //Calculates the velocity as an scalar based on Equation 11/ 12
-                      const double chi = log10(sat.getAreaToMassRatio());
-                      const double mu = factor * chi + offset;
-                      static constexpr double sigma = 0.4;
-                      std::normal_distribution normalDistribution{mu, sigma};
-                      double velocity = std::pow(10, normalDistribution(_randomNumberGenerator));
-
-                      //Transform the scalar velocity into a cartesian vector
-                      auto velocityVector = calculateVelocityVector(velocity);
-                      sat.addEjectionVelocity(velocityVector);
-                  });
+    using namespace util;
+    auto tupleView = _output.getVelocityTuple();
+    std::for_each(std::execution::par, tupleView.begin(), tupleView.end(),
+                  [&](auto &tuple) {
+        //Order in the tuple: 0: A/M | 1: Velocity | 2: Ejection Velocity
+        //Calculates the velocity as a scalar based on Equation 11/ 12
+        const double chi = log10(std::get<0>(tuple));
+        const double mu = factor * chi + offset;
+        static constexpr double sigma = 0.4;
+        std::normal_distribution normalDistribution{mu, sigma};
```

Comment:
```suggestion
        std::normal_distribution<> normalDistribution{mu, sigma};
```

my clang 12 seems to want this here